instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/SpanSuppressors.java

```diff
@@ -63,10 +63,10 @@ public boolean shouldSuppress(Context parentContext, SpanKind spanKind) {
 
   static final class BySpanKey implements SpanSuppressor {
 
-    private final Set<SpanKey> spanKeys;
+    private final SpanKey[] spanKeys;
 
     BySpanKey(Set<SpanKey> spanKeys) {
-      this.spanKeys = spanKeys;
+      this.spanKeys = spanKeys.toArray(new SpanKey[spanKeys.size()]);
```

Comment:
using `new SpanKey[0]` might be a bit more performant on jdk 8 https://shipilev.net/blog/2016/arrays-wisdom-ancients/

Reply:
Happy to change if you like, but note that this is not in the hot path and only happens per-Instrumenter at startup.